### PR TITLE
fix: openssl is compiled twice, once not vendored in linux

### PR DESCRIPTION
**Pull request by xxchan**
It’s broken since the workspace-hack change (#12961).

Openssl-sys is compiled twice on Linux, because features are not unified.

The one for build-dependency doesn’t have openssl-sys feature. It’s depended by tls-native, which is depended by sql-x, which is build-dependency in workspace-hack. (Since we didn't configure hakari to unify target os-specific dependencies, hakari unified it only *partialy*, and this cause them unnecessarily compiled twice.)

Simply put, it’s now a build-dependency so
- Feature is not unified. This can be fixed easily and is done in this PR.
- It will be compiled twice in release mode. Unless #12362, but it’s broken. Not a large problem(? Well, I’m not sure But it has been so). Just slower…

This only happens on linux, but not mac, because native-tls works differently for different os.

Temporary workaround this by manually unify feature in workspace-config to unblock daily pipeline and stuff. Later, we might delete [build-dependencies] for docker/release. Or completely get rid of workspace-hack.

TLDR: After this change, on Linux openssl is compiled once in debug mode, and still twice in release mode. Both once on mac.

I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by xxchan** [comment]
docker build passed

**Review by BugenZhao — APPROVED** [review]
